### ISSUE TITLE
[chore]: use `unicorn/filename-case` eslint rule

### ISSUE THREAD

**Issue by BlackHole1** [netless-io/flat#1080]
Previously, we discussed the specification of file names (see: https://github.com/netless-io/flat/issues/714)

But this is all built on an ego-driven basis and we need to have a rule to help us check this.

> Thanks to @LitoMore  recommendation, we were made aware of the existence of `unicorn/filename-case`.

Rule Link: https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/filename-case.md

The following rules, which I have just come up with, may not be perfect.

```json
"unicorn/filename-case": [
	"error",
	{
		"cases": {
			"kebabCase": true,
			"ignore": [
				"^[A-Z].+\\.tsx$",
				"^README.*\\.md$",
				".*\\.yaml",
				".*\\.yml"
			],
		}
	}
]
```

**Comment by LitoMore** [comment]
We could allow kebab-case and PascalCase both currently.

For markdown files, I think we don't need too much rules for restricting its name. Because we need to append some [language tags](https://www.ietf.org/rfc/rfc1766.txt) (`zhCN`, `zhTW`) to our markdown files.

```js
"unicorn/filename-case": [
	"error",
	{
		"cases": {
			"kebabCase": true,
			"pascalCase": true,
			"ignore": [
				/.*\.md$/,
				/.*\.(yml|yaml)$/,
			],
		}
	}
]
```

**Comment by BlackHole1** [comment]
That feels down to the fact that this rule may not apply to our situation.
How about we redevelop it on top of it?

**Comment by BlackHole1** [comment]
Or we write a script to do this. Because the main rule of eslint is the code specification, not the filename specification. Its an AST we will not be using.

How do you feel? 
cc: @netless-io/developers

**Comment by Cyberhan123** [comment]
> That feels down to the fact that this rule may not apply to our situation. How about we redevelop it on top of it?

yes, you are right. other issue  https://github.com/sindresorhus/eslint-plugin-unicorn/issues/1577 .There seem to be other special problems

**Comment by LitoMore** [comment]
That's what I want to mention. We are putting ourselves into a complicated situation: create another tool.

If we only use kebab-case then we have nothing to worry about.

For `eslint-plugin-unicorn`, if some rules will occur with lots of file changes and are hard to fix then let us disable them or mark them as warnings.

My main point is to make our project rules simple and easy to maintain.

Any thoughts?

**Comment by Cyberhan123** [comment]
> That's what I want to mention. We are putting ourselves into a complicated situation: create another tool.
> 
> If we only use kebab-case then we have nothing to worry about.
> 
> For `eslint-plugin-unicorn`, if some rules will occur with lots of file changes and are hard to fix then let us disable them or mark them as warnings.
> 
> My main point is to make our project rules simple and easy to maintain.
> 
> Any thoughts?

Do not worry, just write an git hook . that's enough

**Comment by LitoMore** [comment]
> Just write a git hook. that's enough.

Let's go for it.
